models/base.py

```diff
@@ -196,8 +202,16 @@ def make_layer_dict(self, *args, **kwargs) -> LayerDictRaw:
     assert name_ctx.maker is self
     name_ctx.is_subnet_ctx = True
     res = self._subnet_func(*args, **kwargs)
-    Copy()(res, name="output")
-    return self._make_layer_dict_from_subnet_ctx(name_ctx)
+    # check if we receive more than one layer as return of a subnet
+    if isinstance(res, (Layer, type(List[Layer]))):
```

Comment:
I don't understand this check? First you want to check for the case that you get a single output as usual.
Also, you do not want to check for `Layer` but `LayerRef`.
```suggestion
    if isinstance(res, LayerRef):
```

Reply:
You are right about the `LayerRef` I think. I am checking for `List` since the input into `from` for `returnn` might also be a list of layer (which by default then are concated). Maybe I am missing a point here but just checking for `LayerRef` will miss these cases, right?

Reply:
This is the return, not the input.

Reply:
Oh you are right, whoops.